test/e2e/seeder/anvil-seeder.js

```diff
@@ -44,42 +50,56 @@ class AnvilSeeder {
     });
 
     if (contractName === SMART_CONTRACTS.NFTS) {
-      const transaction = await walletClient.sendTransaction({
-        from: fromAddress,
-        data: contractConfig.abi.encodeFunctionData('mintNFTs', [1]),
-        to: receipt.contractAddress,
-      });
-      await publicClient.getTransactionReceipt({ hash: transaction.hash });
+      const mintOptions = {
+        address: receipt.contractAddress,
+        abi: contractConfig.abi,
+        functionName: 'mintNFTs',
+        args: [1],
+        account: fromAddress,
+      };
+
+      // Add gasPrice if hardfork is muirGlacier to indicate it's a legacy tx
+      if (hardfork === 'muirGlacier') {
+        mintOptions.gasPrice = 20000;
+      }
+
+      await walletClient.writeContract(mintOptions);
     }
 
     if (contractName === SMART_CONTRACTS.ERC1155) {
-      const transaction = await walletClient.sendTransaction({
-        from: fromAddress,
-        data: contractConfig.abi.encodeFunctionData('mintBatch', [
-          fromAddress,
-          [1, 2, 3],
-          [1, 1, 100000000000000],
-          '0x',
-        ]),
-        to: receipt.contractAddress,
-      });
-      await publicClient.getTransactionReceipt({ hash: transaction.hash });
+      const mintBatchOptions = {
+        address: receipt.contractAddress,
+        abi: contractConfig.abi,
+        functionName: 'mintBatch',
+        args: [fromAddress, [1, 2, 3], [1, 1, 100000000000000], '0x'],
+        account: fromAddress,
+      };
+
+      // Add gasPrice if hardfork is muirGlacier to indicate it's a legacy tx
+      if (hardfork === 'muirGlacier') {
+        mintBatchOptions.gasPrice = 20000;
+      }
+
+      await walletClient.writeContract(mintBatchOptions);
     }
 
     this.storeSmartContractAddress(contractName, receipt.contractAddress);
   }
 
   async transfer(to, value) {
-    const { publicClient, walletClient } = this.provider;
+    const { publicClient, walletClient, testClient } = this.provider;
     const fromAddress = (await walletClient.getAddresses())[0];
 
     const transaction = await walletClient.sendTransaction({
-      from: fromAddress,
+      account: fromAddress,
       value,
       to,
     });
+    await testClient.mine({
+      blocks: 1,
+    });
```

Comment:
to ensure the tx is confirmed, because we could still get the tx hash without it being confirmed below